internal/rbd/nodeserver.go

```diff
@@ -463,6 +452,46 @@ func (ns *NodeServer) stageTransaction(
 	return transaction, err
 }
 
+// resizeNodeStagePath resizes the device if its encrypted and it also resizes
+// the stagingTargetPath if filesystem needs resize.
+func resizeNodeStagePath(ctx context.Context, transaction *stageTransaction, volID, stagingTargetPath string) error {
+	var err error
+	devicePath := transaction.devicePath
+	if transaction.isEncrypted {
+		mapperFile, mapperPath := util.VolumeMapper(volID)
+		// The volume is encrypted, resize an active mapping
+		err = util.ResizeEncryptedVolume(ctx, mapperFile)
+		if err != nil {
+			log.ErrorLog(ctx, "failed to resize device %s, mapper %s: %w",
+				mapperPath, mapperFile, err)
+
+			return status.Errorf(codes.Internal,
+				"failed to resize device %s, mapper %s: %v", mapperPath, mapperFile, err)
+		}
+		// Use mapper device path for fs resize
+		devicePath = mapperPath
+	}
```

Comment:
empty lines between sections of code make things more readable, please use that to please the eyes of your fellow developers